Building on Ubuntu based distros

I'm currently attempting to build on an Ubuntu machine, and struggling with dependencies.

The compilation errors out at 

    ==== Building ygopro (debug) ====
    game.cpp
    ../gframe/game.cpp:11:16: fatal error: io.h: No such file or directory
     #include <io.h>
                ^
    compilation terminated.
    make[1]: *** [../obj/Debug/ygopro/game.o] Error 1
    make: *** [ygopro] Error 2

A quick search turns up libowfat-dev package providing io.h, but after installing that package I get the following errors

```
==== Building ygopro (debug) ====
game.cpp
In file included from /usr/include/io.h:7:0,
                 from ../gframe/game.cpp:11:
/usr/include/uint64.h:11:18: error: conflicting declaration ‘typedef uint64_t uint64’
 typedef uint64_t uint64;
                  ^
In file included from ../gframe/../ocgcore/ocgapi.h:11:0,
                 from ../gframe/config.h:63,
                 from ../gframe/game.cpp:1:
../gframe/../ocgcore/common.h:16:28: error: ‘uint64’ has a previous declaration as ‘typedef long long unsigned int uint64’
 typedef unsigned long long uint64;
                            ^
In file included from /usr/include/io.h:7:0,
                 from ../gframe/game.cpp:11:
/usr/include/uint64.h:12:17: error: conflicting declaration ‘typedef int64_t int64’
 typedef int64_t int64;
                 ^
In file included from ../gframe/../ocgcore/ocgapi.h:11:0,
                 from ../gframe/config.h:63,
                 from ../gframe/game.cpp:1:
../gframe/../ocgcore/common.h:22:19: error: ‘int64’ has a previous declaration as ‘typedef long long int int64’
 typedef long long int64;
                   ^
../gframe/game.cpp: In member function ‘bool ygo::Game::Initialize()’:
../gframe/game.cpp:25:2: error: ‘_finddata_t’ was not declared in this scope
  _finddata_t fdata;
  ^
../gframe/game.cpp:25:14: error: expected ‘;’ before ‘fdata’
  _finddata_t fdata;
              ^
../gframe/game.cpp:63:45: error: ‘fdata’ was not declared in this scope
  fhandle = _findfirst("expansions\\*.cdb", &fdata);
                                             ^
../gframe/game.cpp:63:50: error: ‘_findfirst’ was not declared in this scope
  fhandle = _findfirst("expansions\\*.cdb", &fdata);
                                                  ^
../gframe/game.cpp:67:34: error: ‘_findnext’ was not declared in this scope
   while(_findnext(fhandle, &fdata) != -1) {
                                  ^
../gframe/game.cpp:72:21: error: ‘_findclose’ was not declared in this scope
   _findclose(fhandle);
                     ^
make[1]: *** [../obj/Debug/ygopro/game.o] Error 1
make: *** [ygopro] Error 2
```

So, I assume this is either the wrong package, bad environment setup or further dependencies that I can't work out. Any help would be appreciated.

--- a/gframe/game.cpp
+++ b/gframe/game.cpp
@@ -8,9 +8,10 @@
 #include "duelclient.h"
 #include "netserver.h"
 #include "single_mode.h"
-#include <io.h>
 
-#ifndef WIN32
+#ifdef _WIN32
+#include <io.h>
+#else
 #include <sys/types.h>
 #include <dirent.h>
 #endif
@@ -22,9 +23,12 @@ namespace ygo {
 Game* mainGame;
 
 bool Game::Initialize() {
+#ifdef _WIN32
 	_finddata_t fdata;
 	long fhandle;
 	char fpath[1000] = "./expansions/";
+	char fpath[1000] = "expansions\\";
+#endif
 	srand(time(0));
 	LoadConfig();
 	irr::SIrrlichtCreationParameters params = irr::SIrrlichtCreationParameters();
@@ -60,6 +64,7 @@ bool Game::Initialize() {
 		return false;
 	if(!dataManager.LoadStrings("strings.conf"))
 		return false;
+#ifdef _WIN32
 	fhandle = _findfirst("./expansions/*.cdb", &fdata);
 	if(fhandle != -1) {
 		strcat(fpath, fdata.name);
@@ -71,6 +76,7 @@ bool Game::Initialize() {
 		}
 		_findclose(fhandle);
 	}
+#endif
 	env = device->getGUIEnvironment();
 	numFont = irr::gui::CGUITTFont::createTTFont(env, gameConf.numfont, 16);
 	adFont = irr::gui::CGUITTFont::createTTFont(env, gameConf.numfont, 12);

--- a/gframe/game.cpp
+++ b/gframe/game.cpp
@@ -27,7 +27,6 @@ bool Game::Initialize() {
 	_finddata_t fdata;
 	long fhandle;
 	char fpath[1000] = "./expansions/";
-	char fpath[1000] = "expansions\\";
 #endif
 	srand(time(0));
 	LoadConfig();
@@ -76,6 +75,25 @@ bool Game::Initialize() {
 		}
 		_findclose(fhandle);
 	}
+#else
+        DIR * dir;
+        struct dirent * dirp;
+        const char *foldername = "./expansions/";
+        if((dir = opendir(foldername)) != NULL) {
+	        while((dirp = readdir(dir)) != NULL) {
+		        size_t len = strlen(dirp->d_name);
+		        if(len < 5 || strcasecmp(dirp->d_name + len - 4, ".cdb") != 0)
+			        continue;
+                        char *filepath = (char *)malloc(sizeof(char)*(len + strlen(foldername)));
+                        strncpy(filepath, foldername, strlen(foldername)+1);
+                        strncat(filepath, dirp->d_name, len);
+                        std::cout << "Found file " << filepath << std::endl;
+                        if (!dataManager.LoadDB(filepath))
+	                        std::cout << "Error loading file" << std::endl;
+                        free(filepath);
+	        }
+	        closedir(dir);
+        }
 #endif
 	env = device->getGUIEnvironment();
 	numFont = irr::gui::CGUITTFont::createTTFont(env, gameConf.numfont, 16);

